actions/app_registry.go

```diff
@@ -0,0 +1,22 @@
+package actions
+
+import (
+	"github.com/gobuffalo/buffalo"
+	"github.com/gomods/athens/pkg/user/mongo"
+	"github.com/markbates/goth/gothic"
+)
+
+func addRegistryRoutes(app *buffalo.App) error {
+	cdnGetter := newCDNGetter()
+	mgoStore := mongo.NewMongoUserStore("127.0.0.1:27017")
+	if err := mgoStore.Connect(); err != nil {
+		return err
+	}
+	// serve go-get requests
+	app.Use(GoGet(cdnGetter))
+	auth := app.Group("/auth")
+	auth.GET("/{provider}", buffalo.WrapHandlerFunc(gothic.BeginAuthHandler))
```

Comment:
shouldnt this be in opposite order?
auth + store (*Handler) - for registry
goGet - for proxy (without auth) 
or is auth needed for getting sources? please ping me to explain. my brain is in friday mode

Reply:
not sure what you mean - the middleware intercepts all requests that came from `vgo get` - those can't require authentication